test/index.js

```diff
@@ -743,3 +743,71 @@ describe('transformPhrase', function () {
     expect(function () { Polyglot.transformPhrase({}); }).to.throw(TypeError);
   });
 });
+
+describe('memoize plural type language', function () {
+  function fakeDefaultExport(moduleRelativePath, stubs) {
```

Comment:
is this necessary? I’m not sure we need a test that it’s memoizing; that seems like it should be a transparent implementation detail.

Reply:
I'm removing it although I think that it's useful to test if it doesn't iterate if memoized, it's kind of clumsy way to test the logic but it's the only way to make sure the memoization works I guess.

Reply:
I suppose we could make sure that the function is called 2+ times - ie, exercise the memoization even tho we're not testing for it?

Reply:
😕 The test case that I've delete tested the memoization. The phrase transformation function is called three times with the same locale and we expect the forEach to be called only two times, the first one is done in the constructor.
```
expect(polyglot.t('test_phrase', 0)).to.equal('0 Names');
expect(polyglot.t('test_phrase', 1)).to.equal('1 Name');
expect(polyglot.t('test_phrase', 2)).to.equal('2 Names');
expect(forEachCalls.length).to.equal(3);
expect(forEachCalls[0][0]).to.eql({ test_phrase: '%{smart_count} Name |||| %{smart_count} Names' }); // this one is called in the constructor
expect(forEachCalls[1][0]).to.eql({ germanLike: ['x1'] });
expect(forEachCalls[2][0]).to.eql(['x1']);
```
Without the memoization the test would fail as the forEach gets called 7+ times.

Reply:
The difference is that we shouldn't be testing "the forEach is called" - we should just ensure that the same values are returned after multiple calls.

Memoization is supposed to be a completely invisible optimization, which means it shouldn't be directly tested for.

Reply:
Ok, got it, it makes sense and thanks for the explanation.

Reply:
np, sorry for the delayed reply